src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -787,15 +810,23 @@ private void printToPathGraphics(    final PeekGraphics graphics, // Always an a
         // This is called from the native side.
         Runnable r = new Runnable() { public void run() {
             try {
+                double scaleX = getXRes()/OS_DEVICE_DPI;
+                double scaleY = getYRes()/OS_DEVICE_DPI;
                 SurfaceData sd = CPrinterSurfaceData.createData(page, context); // Just stores page into an ivar
                 if (defaultFont == null) {
                     defaultFont = new Font("Dialog", Font.PLAIN, 12);
                 }
                 Graphics2D delegate = new SunGraphics2D(sd, Color.black, Color.white, defaultFont);
-
+                if (scaleX != 1 && scaleY != 1) {
+                    ((SunGraphics2D) delegate).setDevClip(0, 0,
+                            (int)(page.getWidth() * scaleX), (int)(page.getHeight() * scaleY));
+                }
                 Graphics2D pathGraphics = new CPrinterGraphics(delegate, printerJob); // Just stores delegate into an ivar
-                Rectangle2D pageFormatArea = getPageFormatArea(page);
+                Rectangle2D pageFormatArea = getPageFormatArea(page, scaleX, scaleY);
```

Comment:
I don't think this is right. scaleX is presumably the un-rotated X resolution, but these PageFormat methods take into account the rotation.
So if you have a printer that has a resolution of x=300dpi, y=600dpi, and you go into landscape mode, then the scales will be the wrong way around.